extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultHealthCheck.java

```diff
@@ -40,51 +38,62 @@ public HashicorpVaultHealthCheck(HashicorpVaultClient client, Monitor monitor) {
     @Override
     public HealthCheckResult get() {
 
-        HealthResponse response;
-        HealthCheckResult result;
-        try {
-            response = client.getHealth();
-        } catch (EdcException e) {  // can be thrown by the client, e.g. on JSON parsing error, etc.
-            var exceptionMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "EdcException: " + e.getMessage(), "");
-            monitor.severe(exceptionMsg, e);
-            return HealthCheckResult.failed(exceptionMsg).forComponent(COMPONENT_NAME);
+        var healthCheckResponseResult = client.doHealthCheck();
+        if (healthCheckResponseResult.failed()) {
+            monitor.severe(HEALTH_CHECK_ERROR);
+            return HealthCheckResult
+                    .failed(healthCheckResponseResult.getFailureMessages())
+                    .forComponent(HashicorpVaultExtension.NAME);
```

Comment:
not the same thing. The extension's name should be separate from the health component.